src/webgpu/api/validation/createBindGroupLayout.spec.ts

```diff
@@ -68,101 +70,89 @@ g.test('visibility_and_dynamic_offsets')
     }, !success);
   });
 
-g.test('min_buffer_binding_size')
+g.test('non_required_parameter_combined_with_type_separately')
   .params(
     params()
       .combine(poptions('type', kBindingTypes))
-      .combine(poptions('minBufferBindingSize', [undefined, 0, 4]))
+      .combine([
+        ...poptions('minBufferBindingSize', [undefined, 0, 4]),
+        ...poptions('textureComponentType', [undefined, ...kTextureComponentTypes]),
+        ...poptions('multisampled', [undefined, false, true]),
+        ...poptions('viewDimension', [undefined, ...kTextureViewDimensions]),
+        ...poptions('storageTextureFormat', [undefined, ...kTextureFormats]),
+      ])
   )
-  .fn(async t => {
-    const { type, minBufferBindingSize } = t.params;
-
-    let success = false;
-    if (
+  .fn(t => {
+    const {
+      type,
+      minBufferBindingSize,
+      textureComponentType,
+      multisampled,
+      viewDimension,
+      storageTextureFormat,
+    } = t.params;
+
+    const minBufferBindingSizeValid =
       minBufferBindingSize === undefined ||
       minBufferBindingSize === 0 ||
-      type in kBufferBindingTypeInfo
-    ) {
-      success = true;
-    }
-
-    t.expectValidationError(() => {
-      t.device.createBindGroupLayout({
-        entries: [{ binding: 0, visibility: GPUShaderStage.COMPUTE, type, minBufferBindingSize }],
-      });
-    }, !success);
-  });
-
-g.test('view_dimension')
-  .params(
-    params()
-      .combine(poptions('type', kBindingTypes))
-      .combine(poptions('viewDimension', [undefined, ...kTextureViewDimensions]))
-  )
-  .fn(async t => {
-    const { type, viewDimension } = t.params;
-
-    const success = viewDimension === undefined || type in kTextureBindingTypeInfo;
-
-    t.expectValidationError(() => {
-      t.device.createBindGroupLayout({
-        entries: [{ binding: 0, visibility: GPUShaderStage.COMPUTE, type, viewDimension }],
-      });
-    }, !success);
-  });
-
-g.test('texture_component_type')
-  .params(
-    params()
-      .combine(poptions('type', kBindingTypes))
-      .combine(poptions('textureComponentType', [undefined, ...kTextureComponentTypes]))
-  )
-  .fn(async t => {
-    const { type, textureComponentType } = t.params;
+      type in kBufferBindingTypeInfo;
+    const textureComponentTypeValid =
+      textureComponentType === undefined || kBindingTypeInfo[type].resource === 'sampledTex';
+    const multisampledValid =
+      multisampled === false || kBindingTypeInfo[type].resource === 'sampledTex';
```

Comment:
Do we need to change this to "multisampled  === undefined || multisampled === false || kBindingTypeInfo[type].resource === 'sampledTex';"? I remember somewhere we mentioned that when we reach here, the undefined multisampled has already been  assigned to false. 

So another question is that other parameters like minBufferBindingSize, viewDimensionValid and textureComponentType have default value, but they are assigned to their default value at this point. Why? And these parameters are not assigned the default value at its definition in spec: dictionary GPUBindGroupLayoutEntry { ... } at https://gpuweb.github.io/gpuweb/#bind-group-layout-creation. Any reasons for the differences between multisampled/hasDynamicOffset vs minBufferBindingSize/viewDimensionValid/textureComponentType? Or it is just because the spec is incomplete?

Reply:
Personally, I think it's better to make those two boolean parameters (and minBufferBindingSize) consistent with the others. Otherwise, it looks a little bit weird and inconsistent. I created a PR at https://github.com/gpuweb/gpuweb/pull/949. PTAL.